### PR TITLE
Splash Attention creates a NumPy mask inside jax.ensure_compile_time_eval()

### PR DESCRIPTION
**Pull request by ds-hwang**
This is because mask.mask can use jnp ops (e.g., jnp.logical_and). If that happens during tracing, it produces tracer objects, which can’t be converted to NumPy arrays.

To fix this, we restore the use of jax.ensure_compile_time_eval() around the custom mask creation.

### CODE REVIEW

**Comment by ds-hwang** [comment]
@markblee Could you review? From 1161

**Review by markblee — APPROVED** [review]
